### PR TITLE
Data deserialize

### LINKED ISSUE #4650: Data: Error when trying to commit
Files for one table look ok but "unpack not ok" error when trying to commit to repo

```json
/data//mbtools/cmds.config.json 
{
  "name": "/MBTOOLS/CMDS",
  "type": "TABU",
  "where": []
}
```
```json
/data//mbtools/cmds.tabu.json 
[
  {
    "command": "CALC",
    "shortcut": "=",
    "clsname": "/MBTOOLS/CL_COMMAND__CALC",
    "description": "Calculate"
  },
  { 
    "...": "..."
  }
]
```

```abap
ZCL_ABAPGIT_GIT_TRANSPORT->RECEIVE_PACK 
lv_line = `66e5c990075279e729a73f48a0d0be119b90ec1b ccb967b390c61924a761c676aec68f522b1d8614 refs/heads/main# report-status#`

lv_string = '0029#0024unpack index-pack abnormal exit#001eng refs/heads/main failed#00000000'
```

image

### PR DESCRIPTION
**Pull request by mbtools**
Continuing on #3441 

- Fixes issue with namespaced tables (closes #4650)
- Adds deserialize of table content during pull but does NOT persist changes to database
- Some refactoring to keep symmetry between serialize and deserialize
- Adds check to prevent mass data serialization (currently max 10k records)
- Adds unit tests for data_util

`zcl_abapgit_repo->deserialize` now contains the following call to deserialize "data":

```abap
    " Deserialize data (no save to database, just test for now)
    lt_result = zcl_abapgit_data_factory=>get_deserializer( )->deserialize(
      ii_config  = get_data_config( )
      it_files   = get_files_remote( ) )
```

### CODE REVIEW

File: src/data/zcl_abapgit_data_config.clas.abap [hunk 1]
```diff
@@ -24,20 +22,20 @@ ENDCLASS.
...
 
 
   METHOD dump.
 
-    DATA lo_ajson TYPE REF TO zcl_abapgit_ajson.
-    DATA lx_ajson TYPE REF TO zcx_abapgit_ajson_error.
-
+    DATA:
+      lo_ajson TYPE REF TO zcl_abapgit_ajson,
+      lx_ajson TYPE REF TO zcx_abapgit_ajson_error.
```
Comment (larshp) [inline src/data/zcl_abapgit_data_config.clas.abap:32]: the previous code follows https://github.com/SAP/styleguides/blob/main/clean-abap/CleanABAP.md#do-not-chain-up-front-declarations 🤷‍♂️  so someone will probably change it back again

File: src/data/zcl_abapgit_data_deserializer.clas.abap [hunk 2]
```diff
@@ -41,23 +67,117 @@ CLASS ZCL_ABAPGIT_DATA_DESERIALIZER IMPLEMENTATION.
...
+
+    lv_tabname = iv_name.
+
+    ASSIGN ir_data->* TO <lg_tab>.
+
+    LOOP AT it_where INTO lv_where.
+      DELETE FROM (lv_tabname) WHERE (lv_where) ##SUBRC_OK.
+    ENDLOOP.
+    IF lines( it_where ) = 0.
+      CALL FUNCTION 'DB_TRUNCATE_TABLE'
```
Comment (larshp) [inline src/data/zcl_abapgit_data_deserializer.clas.abap:150]: I dont think we need this, yea, truncate is faster, but for 10k records? 🤷‍♂️ rather just keep it simple and avoid calling functionality we are not in control of

File: src/data/zcl_abapgit_data_deserializer.clas.abap [hunk 3]
```diff
@@ -41,23 +67,117 @@ CLASS ZCL_ABAPGIT_DATA_DESERIALIZER IMPLEMENTATION.
...
+      ENDIF.
+    ENDIF.
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_data_deserializer~deserialize.
+
+* as a default, this method does NOT change any database tables.
+* you have to explicitly set iv_persist = abap_true to save the data
```
Comment (larshp) [inline src/data/zcl_abapgit_data_deserializer.clas.abap:173]: https://github.com/SAP/styleguides/blob/main/clean-abap/CleanABAP.md#split-method-instead-of-boolean-input-parameter

**Review by larshp — APPROVED** [review]
few comments, but okay to merge 👍

File: src/data/zcl_abapgit_data_config.clas.abap (hunk 1 above)
Comment (mbtools) [reply src/data/zcl_abapgit_data_config.clas.abap:32]: you can tell, I disagree with the rule ;-). the zig-zag and repeating of `DATA` makes it harder to parse with the eye and then there's PP to align this block automatically. if you prefer the "clean" style, i will change it. (should go on here too https://docs.abapgit.org/development/guidelines.html)

Comment (mbtools) [reply src/data/zcl_abapgit_data_config.clas.abap]: changed back

**Comment by mbtools** [comment]
basic tests - also with `actualize` - look good.

**Comment by larshp** [comment]
cool, merging

**Pull request merged**